Add extra byte to BaseHighLimitRecordFormat header.

maxsumrall opened the pull request:
In the future if we want to add extra bits to the header,
we would have to rewrite all the records to account for this.
By reserving this space now, we give ourselves the ability to
start using those bits in the future without migrating.

Commit:
Add extra byte to BaseHighLimitRecordFormat header.

In the future if we want to add extra bits to the header,
we would have to rewrite all the records to account for this.
By reserving this space now, we give ourselves the ability to
start using those bits in the future without doing a complicated
migration.

davidegrohmann commented:
@maxsumrall no tests?

chrisvest commented:
@maxsumrall How does this influence the likelihood of e.g. relationship records spilling to two record units?

Commit:
Extra test asserting second header byte value.

maxsumrall commented:
@chrisvest This will increase the likelihood since we have 1 less byte now to use for writing data. I'm not sure how to calculate precisely how more often we will spill now.

Commit:
Increment high limit format to E.H.1

Commit:
Fixed record splitting in high limit test format.

The code splitting records across two units prematurely
for testing purposes needs to be changed to account for
the extra head byte.

spacecowboy commented on enterprise/kernel/src/test/java/org/neo4j/kernel/impl/store/format/highlimit/BaseHighLimitRecordFormatTest.java:
```diff
@@ -0,0 +1,64 @@
+        nodeRecord.setNextRel( Integer.MAX_VALUE );
+        nodeRecord.setNextProp( Integer.MAX_VALUE );
+        IntStoreHeader intStoreHeader = new IntStoreHeader( 42 );
+        int recordSize = nodeRecordFormat.getRecordSize( intStoreHeader);
```
missing space here, and on next line :)

spacecowboy commented on enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/BaseHighLimitRecordFormat.java:
```diff
@@ -80,7 +80,7 @@
         extends BaseOneByteHeaderRecordFormat<RECORD>
 {
-    private static final int HEADER_BYTE = Byte.BYTES;
+    private static final int HEADER_SIZE = Byte.BYTES + Byte.BYTES;
```
why not `2 * Byte.BYTES`? :)

spacecowboy commented on enterprise/kernel/src/test/java/org/neo4j/kernel/impl/store/format/highlimit/HighLimitWithSmallRecords.java:
```diff
@@ -35,9 +35,14 @@
+     * This causes problems with splitting across values in this small records high limits format used in tests.
+     * The +1 to the following 3 record sizes will cause it to not split across a value.
+     */
+    private static final int NODE_RECORD_SIZE = (NodeRecordFormat.RECORD_SIZE / 2) + 1;
```
Missing spaces all around here :)

Commit:
Code Style fixes

spacecowboy commented:
Holding off on merging, until we have discussed the change in more detail

tinwelint commented:
I'm closing this for now because there's no appealing reason to merge at this point in time.
